src/wallet/bdk_wallet/error.rs

```diff
@@ -0,0 +1,137 @@
+    BDKErrors(BDKErrors),
+}
+
+/// Enum for handling BDK-related errors.
+#[derive(Debug)]
+pub enum BDKErrors {
+    /// Errors related to the parsing and usage of passed-in descriptor(s).
+    Descriptor(bdk_wallet::descriptor::error::Error),
+    /// Errors that can happen while extracting and manipulating policies.
+    Policy(bdk_wallet::descriptor::policy::PolicyError),
+    /// The error type when constructing a fresh [`Wallet`].
+    New(bdk_wallet::wallet::NewError),
+    /// The error type when loading a [`Wallet`] from a [`ChangeSet`].
+    Load(bdk_wallet::wallet::LoadError),
+    /// Error type for when we try load a [`Wallet`] from persistence and creating it if non-existent.
+    NewLoad(bdk_wallet::wallet::NewOrLoadError),
+    /// An error that may occur when applying a block to [`Wallet`].
+    ApplyBlock(bdk_wallet::wallet::ApplyBlockError),
+}
+
+/// Implementations for Wallet Errors.
+impl From<std::io::Error> for WalletError {
+    fn from(value: std::io::Error) -> Self {
+        Self::File(value)
+    }
+}
+
+impl From<serde_cbor::Error> for WalletError {
+    fn from(value: serde_cbor::Error) -> Self {
+        Self::Cbor(value)
+    }
+}
+
+impl From<bitcoind::bitcoincore_rpc::Error> for WalletError {
+    fn from(value: bitcoind::bitcoincore_rpc::Error) -> Self {
+        Self::Rpc(value)
+    }
+}
+
+impl From<bitcoin::bip32::Error> for WalletError {
+    fn from(value: bitcoin::bip32::Error) -> Self {
+        Self::BIP32(value)
+    }
+}
+
+impl From<bip39::Error> for WalletError {
+    fn from(value: bip39::Error) -> Self {
+        Self::BIP39(value)
+    }
+}
+
+impl From<crate::protocol::error::ContractError> for WalletError {
+    fn from(value: crate::protocol::error::ContractError) -> Self {
+        Self::Contract(value)
+    }
+}
+
+impl From<crate::wallet::FidelityError> for WalletError {
+    fn from(value: crate::wallet::FidelityError) -> Self {
+        Self::Fidelity(value)
+    }
+}
+
+impl From<bitcoin::blockdata::locktime::absolute::ConversionError> for WalletError {
+    fn from(value: bitcoin::blockdata::locktime::absolute::ConversionError) -> Self {
+        Self::Locktime(value)
+    }
+}
+
+impl From<bitcoin::secp256k1::Error> for WalletError {
+    fn from(value: bitcoin::secp256k1::Error) -> Self {
+        Self::Secp(value)
+    }
+}
+
+/// Implementations for BDK Errors.
+impl From<bdk_wallet::descriptor::DescriptorError> for WalletError {
+    fn from(value: bdk_wallet::descriptor::DescriptorError) -> Self {
+        Self::BDKErrors(BDKErrors::Descriptor(value))
+    }
+}
+
+impl From<bdk_wallet::descriptor::policy::PolicyError> for WalletError {
+    fn from(value: bdk_wallet::descriptor::policy::PolicyError) -> Self {
+        Self::BDKErrors(BDKErrors::Policy(value))
+    }
+}
+
+impl From<bdk_wallet::wallet::NewError> for WalletError {
+    fn from(value: bdk_wallet::wallet::NewError) -> Self {
+        Self::BDKErrors(BDKErrors::New(value))
+    }
+}
```

Comment:
It will be much cleaner code if you modularize these Form impls. 

- `bdk/error.rs` -> Defines all the `From<T> for BDKError`. 
- `wallet/error.rs` -> just add `<From<BdkError> for WalletError`.